File: app/src/main/java/com/exercise/davismiyashiro/popularmovies/data/local/MoviesDb.java
package com.exercise.davismiyashiro.popularmovies.data.local;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

import com.exercise.davismiyashiro.popularmovies.data.MovieDetails;

/**
 * Created by Davis Miyashiro.
 */
@Database(entities = {MovieDetails.class}, version = 1, exportSchema = false)
public abstract class MoviesDb extends RoomDatabase {

    public static final String DATABASE_NAME = "moviesdatabase.db";
    public static final String TABLE_NAME = "movies";

    public abstract MoviesDao moviesDao();

    private static MoviesDb INSTANCE;

    public synchronized static MoviesDb getDatabase (final Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                    MoviesDb.class, DATABASE_NAME)
                    .build();
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/exercise/davismiyashiro/popularmovies/data/local/MoviesDb.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Davis Miyashiro
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.exercise.davismiyashiro.popularmovies.data.local

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import android.content.Context

import com.exercise.davismiyashiro.popularmovies.data.MovieDetails

const val TABLE_NAME = "movies"
/**
 * Created by Davis Miyashiro.
 */
@Database(entities = [MovieDetails::class], version = 1, exportSchema = false)
abstract class MoviesDb : RoomDatabase() {

    abstract fun moviesDao(): MoviesDao

    companion object {

        private const val DATABASE_NAME = "moviesdatabase.db"

        private var INSTANCE: MoviesDb? = null

        @Synchronized
        fun getDatabase(context: Context): MoviesDb? {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(context.applicationContext,
                        MoviesDb::class.java, DATABASE_NAME)
                        .build()
            }
            return INSTANCE
        }
    }
}
